library/alloc/src/collections/vec_deque.rs

```diff
@@ -1090,11 +1090,13 @@ impl<T> VecDeque<T> {
         let len = self.len();
         let start = match range.start_bound() {
             Included(&n) => n,
-            Excluded(&n) => n + 1,
+            // Detect overflow with bounds checking code below.
+            // Using a saturated index is fine, because usize::MAX index is never valid.
```

Comment:
I think it can be valid for a deque of ZSTs?

Reply:
Currently this is not possible and I think unlikely to change. The capacity is a power of two (for zero sized types MAXIMUM_ZST_CAPACITY is used) so maximum size is much smaller than that. Additionally one element is always left vacant.